src/main/java/org/jenkinsci/plugins/docker/commons/credentials/DockerRegistryToken.java

```diff
@@ -62,12 +65,20 @@ public String getToken() {
     }
 
     /**
+     * @deprecated Call {@link #newKeyMaterialFactory(URL, VirtualChannel, Launcher, TaskListener)}
+     */
+    @Deprecated
+    public KeyMaterialFactory newKeyMaterialFactory(final URL endpoint, @Nonnull VirtualChannel target) throws InterruptedException, IOException {
+        return newKeyMaterialFactory(endpoint, target, null, TaskListener.NULL);
+    }
+
+    /**
      * Makes the credentials available locally and returns {@link KeyMaterialFactory} that gives you the parameters
      * needed to access it.
      *
      * This is done by inserting the token into {@code ~/.dockercfg}
      */
-    public KeyMaterialFactory newKeyMaterialFactory(final URL endpoint, @Nonnull VirtualChannel target) throws InterruptedException, IOException {
+    public KeyMaterialFactory newKeyMaterialFactory(final @Nonnull URL endpoint, @Nonnull VirtualChannel target, @CheckForNull Launcher launcher, final @Nonnull TaskListener listener) throws InterruptedException, IOException {
```

Comment:
Not currently using the `launcher`, but left as an option should we wish to call `docker login`—though a `close` implementation ([JENKINS-28702](https://issues.jenkins-ci.org/browse/JENKINS-28702)) could not run `docker logout` using a `Launcher` for serialization reasons, unless the `KeyMaterial` API were also modified to accept a `Launcher` and `TaskListener`.